src/System.Net.Http/tests/FunctionalTests/HttpRequestMessageTest.cs

```diff
@@ -251,6 +251,34 @@ public async Task HttpRequest_BodylessMethod_NoContentLength(string method)
             }
         }
 
+        [Fact]
+        public async Task HttpRequest_BodylessMethod_LargeContentLength()
+        {
+            using (HttpClient client = CreateHttpClient())
+            {
+                await LoopbackServer.CreateServerAsync(async (server, uri) =>
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Head, uri);
+
+                    Task<HttpResponseMessage> requestTask = client.SendAsync(request);
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        // Content-Length greater than 2GB.
+                        var response = LoopbackServer.GetConnectionCloseResponse(
+                            HttpStatusCode.OK, "Content-Length: 2167849215\r\n\r\n");
+                        await connection.SendResponseAsync(response);
+
+                        await requestTask;
+                    });
+
+                    var result = requestTask.Result;
```

Comment:
As previously commented, this should be using await, not .Result, and it should dispose of the HttpResponseMessage for good hygiene:
```C#
using (HttpResponseMessage result = await requestTask)
{
    ...
}
```